crates/data-plane-controller/tests/snapshots/test_initial_convergence_and_release__test.snap

```diff
@@ -139,9 +273,63 @@ expression: trace.lock().unwrap().as_slice()
       "pulumi refresh --stack test-stack --diff --non-interactive --skip-preview --yes --expect-no-changes"
     ]
   },
+  {
+    "Cmd": [
+      "git-clean",
+      "git clean --force"
+    ]
+  },
+  {
+    "Cmd": [
+      "git-fetch",
+      "git fetch"
+    ]
+  },
+  {
+    "Cmd": [
+      "git-checkout",
+      "git checkout --detach --force --quiet origin/main"
+    ]
+  },
   {
     "Status": "Idle"
   },
+  {
+    "Cmd": [
+      "git-clean",
+      "git clean --force"
+    ]
+  },
+  {
+    "Cmd": [
+      "git-fetch",
+      "git fetch"
+    ]
+  },
+  {
```

Comment:
I ... think this should be fine? But it do still have a latent concern over the sheer number of git operations / fetches in particular that we're doing.

Do you see an opportunity to reduce that at all?

Reply:
One possible way to reduce load of these operations is using `git sparse-checkout` so for ops repository we only checkout the one file we need, and subsequent checkouts will only get updates to that file and not the whole ops repository.

I would prefer keeping the validation before and after all operations to avoid bugs where we do not validate state in specific cases... but perhaps we can make the checkout less intense

Reply:
Using `git sparse-checkout` means _more_ commands to run and more complexity in the code, and I don't know how much it's going to help past the first `git clone` that will be slightly cheaper, otherwise subsequent checkouts already are not that expensive

Reply:
One other possibility is downloading the file using HTTP GET with a GitHub auth header instead of checking out the repository, that is probably somewhat cheaper compared to a git clone, but subsequent git checkouts are probably not much more expensive either